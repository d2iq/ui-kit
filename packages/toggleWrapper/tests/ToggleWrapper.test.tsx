import React from "react";
import { mount } from "enzyme";
import serializer from "@emotion/jest";
import toJson from "enzyme-to-json";

import { ToggleWrapper } from "../";

expect.addSnapshotSerializer(serializer);

describe("ToggleWrapper", () => {
  it("renders", () => {
    const component = mount(
      <ToggleWrapper isActive={true}>
        {({ isActive }) => <div>{`isActive? ${isActive}`}</div>}
      </ToggleWrapper>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders as a radio input", () => {
    const component = mount(
      <ToggleWrapper type="radio" isActive={true}>
        {({ isActive }) => <div>{`isActive? ${isActive}`}</div>}
      </ToggleWrapper>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("calls onFocus prop when the input gets focus", () => {
    const focusFn = jest.fn();
    const component = mount(
      <ToggleWrapper isActive={true} onFocus={focusFn}>
        {({ isActive }) => <div>{`isActive? ${isActive}`}</div>}
      </ToggleWrapper>
    );

    expect(focusFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(focusFn).toHaveBeenCalled();
  });
  it("calls onBlur prop when the input loses focus", () => {
    const blurFn = jest.fn();
    const component = mount(
      <ToggleWrapper isActive={true} onBlur={blurFn}>
        {({ isActive }) => <div>{`isActive? ${isActive}`}</div>}
      </ToggleWrapper>
    );

    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("blur");
    expect(blurFn).toHaveBeenCalled();
  });
});
