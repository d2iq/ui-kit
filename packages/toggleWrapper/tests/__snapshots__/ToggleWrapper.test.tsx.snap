// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleWrapper renders 1`] = `
.emotion-0 {
  display: block;
}

.emotion-1 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

<ToggleWrapper
  isActive={true}
  type="checkbox"
>
  <label
    className="emotion-0"
  >
    <input
      aria-checked={true}
      checked={true}
      className="emotion-1"
      data-cy="toggleWrapper-input"
      onBlur={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <div>
      isActive? true
    </div>
  </label>
</ToggleWrapper>
`;

exports[`ToggleWrapper renders as a radio input 1`] = `
.emotion-0 {
  display: block;
}

.emotion-1 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

<ToggleWrapper
  isActive={true}
  type="radio"
>
  <label
    className="emotion-0"
  >
    <input
      aria-checked={true}
      checked={true}
      className="emotion-1"
      data-cy="toggleWrapper-input"
      onBlur={[Function]}
      onFocus={[Function]}
      type="radio"
    />
    <div>
      isActive? true
    </div>
  </label>
</ToggleWrapper>
`;
